src/app/view/web/WebChessView.ts

```diff
@@ -0,0 +1,143 @@
+import { Chessboard } from './Chessboard';
+import { ChessBoardView } from '../ChessBoardView';
+import { ViewEventBus } from '../events/ViewEventBus';
+import { ViewEvent } from '../events/ViewEvent';
+import { SquareWasClicked } from '../events/SquareWasClicked';
+import { PromotionChosenPiece } from '../events/PromotionChosenPiece';
+import { PiecesBoardPositions } from '../Types';
+
+export class WebChessView implements ChessBoardView {
+  constructor(private readonly viewEventBus: ViewEventBus, private readonly parent: HTMLElement = document.body) {}
+
+  showChessBoard(piecesPositions: PiecesBoardPositions): void {
+    const chessboard: Chessboard = new Chessboard('chessBoardId', 'chessboard', (position) =>
+      this.viewEventBus.publish(new SquareWasClicked(position)),
+    );
+    this.parent.appendChild(chessboard.createBoard());
+
+    this.renderPiecesOnBoard(piecesPositions);
+  }
+
+  listenOn<EventType extends ViewEvent = ViewEvent>(eventType: EventType['eventType'], reaction: (event: EventType) => void): void {
+    this.viewEventBus.listenOn(eventType, reaction);
+  }
+
+  showSelectedPiece(id: string): void {
+    if (this.parent.querySelector(`#${id}`)?.hasChildNodes()) {
+      this.parent.querySelector(`#${id}`)?.classList.add('square--selected');
+    }
+  }
+
+  hideSelection(): void {
+    this.parent.querySelectorAll('.square')?.forEach((square) => {
+      square.classList.remove('square--selected');
+    });
+  }
+
+  showAvailableMoves(squaresToHighlight: string[]): void {
+    squaresToHighlight.forEach((square) => {
+      this.parent.querySelector(`#${square}`)?.classList.add('square--possibleMove');
+    });
+  }
+
+  hideAllAvailableMoves(): void {
+    this.parent.querySelectorAll('.square')?.forEach((square) => {
+      square.classList.remove('square--possibleMove');
+    });
+  }
+
+  movePiece(squareFrom: string, squareTo: string): void {
+    const divFrom = this.parent.querySelector(`#${squareFrom}`);
+    const divTo = this.parent.querySelector(`#${squareTo}`);
+    const pieceImage = divFrom?.firstChild;
+
+    if (pieceImage) {
+      divFrom?.removeChild(pieceImage);
+      divTo?.appendChild(pieceImage);
+    }
+  }
+
+  capturePiece(onSquare: string): void {
+    const divFrom = this.parent.querySelector(`#${onSquare}`);
+    const pieceImage = divFrom?.firstChild;
+
+    if (pieceImage) {
+      divFrom?.removeChild(pieceImage);
+    }
+  }
+
+  afterPromotionPiece(onSquare: string, piece: string, side: string) {
+    const divFrom = this.parent.querySelector(`#${onSquare}`);
+    const pieceImage = divFrom?.firstChild as HTMLImageElement;
+
+    if (pieceImage) {
+      pieceImage.src = `static/img/pieces/${side.toLowerCase()}-${piece.toLowerCase()}.svg`;
```

Comment:
Dobrym pomysłem mogłoby być przechowywanie początka ścieżki (static/img/pieces/) w zmiennej globalnej, na wypadek zmian w strukturze katalogów.
W tym przypadku występuje to także w 83 linijce